### PR TITLE
Remote tokenization service support

### PR DESCRIPTION
**Pull request by irina060981**
For #42 

- added AlpheiosRemoteTokenizer
- updated alignment tokenize workflow to async/await
- added notifications to the bar
- added parameters for the tokenization to application settings (index.html)
- updated descriptions and tests

### CODE REVIEW

File: src/lib/tokenizers/alpheios-remote-tokenizer.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    const out = []
+
+    for (var i = 0; i < text.length; i++) {
+      out[i] = text.charCodeAt(i)
+    }
+
+    const outBinaryArray = new Uint8Array(out)
+
+    return new Blob([outBinaryArray], {
+      type: 'text/plain'
```
Comment (balmas) [inline src/lib/tokenizers/alpheios-remote-tokenizer.js:51]: question: is this only necessary for plain text input in the textarea? /Does posting of an uploaded file work without this?

File: src/lib/tokenizers/alpheios-remote-tokenizer.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     * @param {String} idPrefix - prefix for creating tokens idWord
+     * @returns {[Objects]} - array of token-like objects, would be converted to Tokens outside
+     */
+  static async tokenize (docSource, idPrefix, tokenizeParams) {
+    var textBlob = this.convertStringToBinaryBlob(docSource.text)
+    const fetchOptions = Object.assign({
+      lang: 'lat',
+      textType: 'text',
+      segments: 'singleline'
+    }, tokenizeParams)
```
Comment (balmas) [inline src/lib/tokenizers/alpheios-remote-tokenizer.js:16]: can wait for a later pr but eventually options available should come from the service apidoc

**Review by balmas — APPROVED** [review]
this looks ok - with next pr can we make the remote service the default (or make a temporary ui switch available for testing?)  See also note and question in comments. tx.

**Comment by irina060981** [comment]
>this looks ok - with next pr can we make the remote service the default (or make a temporary ui switch available for testing?) See also note and question in comments. tx.

It is already done:
```
export default class AlignedController {
...
  get defaultTokenizer () {
    return 'alpheiosRemoteTokenizer'
  }
```
and we could update it inside Application initialization

```
          new window.AlignmentEditor.AppController({
            appId: 'alpheios-alignment-editor',
            theme: 'v1-theme',
            tokenizeParams: {
              // tokenizer: 'simpleLocalTokenizer',
              // segments: 'doubline'
            }
          }).init()
```

File: src/lib/tokenizers/alpheios-remote-tokenizer.js (hunk 1 above)
Comment (irina060981) [reply src/lib/tokenizers/alpheios-remote-tokenizer.js:61]: I am not sure that I understood the question correctly 
We send text from textarea, it is always String - so I need to convert it to binary.

Do you mean here , that we should create a file? a send it as a windows.fetch? But I do the same.
What is your suggestion?

File: src/lib/tokenizers/alpheios-remote-tokenizer.js (hunk 2 above)
Comment (irina060981) [reply src/lib/tokenizers/alpheios-remote-tokenizer.js]: Do you mean that we should do everytime an additional request to the remote service for the default settings?
lang and textType are not default
now we support only text,  but `textType` and `lang` are the properties of the source text
But here are the mistake with lang (defined always as lat), I fixed it

segments are not obligatory - I simply leave it here after testing doubline overwrite from the applications parameters. 
Would remove.

**Comment by irina060981** [comment]
Also I added small changes to handle with 500Error from remote tokenization service.
https://github.com/alpheios-project/alignment-editor-new/issues/50
https://github.com/alpheios-project/alignment-editor-new/issues/54

File: src/lib/tokenizers/alpheios-remote-tokenizer.js (hunk 1 above)
Comment (balmas) [reply src/lib/tokenizers/alpheios-remote-tokenizer.js:60]: Upload allows you to upload a text file, right? That's what I'm referring to

File: src/lib/tokenizers/alpheios-remote-tokenizer.js (hunk 2 above)
Comment (balmas) [reply src/lib/tokenizers/alpheios-remote-tokenizer.js]: The user should be able to specify what settings to use for tokenization.  E.g only they know if they are wanting a single segment or multiple, how segments are specified (singe/double line for text, element names for tei, etc)  We may make additional options available as we go. So the client (alignment editor) needs to serve as a proxy to the tokenization service - showing the user the options and passing them on to the service.  I would like not to have the options duplicated in the client, but instead retrieved from the service api doc.

See for example how we presented this sort of thing with the v1 editor in perseids:

dd75bc68-2552-47f2-8d46-935d9036259e

**Comment by balmas** [comment]
> > this looks ok - with next pr can we make the remote service the default (or make a temporary ui switch available for testing?) See also note and question in comments. tx.
> 
> It is already done:
> 
> ```
> export default class AlignedController {
> ...
>   get defaultTokenizer () {
>     return 'alpheiosRemoteTokenizer'
>   }
> ```
> 
> and we could update it inside Application initialization
> 
> ```
>           new window.AlignmentEditor.AppController({
>             appId: 'alpheios-alignment-editor',
>             theme: 'v1-theme',
>             tokenizeParams: {
>               // tokenizer: 'simpleLocalTokenizer',
>               // segments: 'doubline'
>             }
>           }).init()
> ```

Ok it looked to me like the simple local tokenizer was set in the index.html

File: src/lib/tokenizers/alpheios-remote-tokenizer.js (hunk 1 above)
Comment (irina060981) [reply src/lib/tokenizers/alpheios-remote-tokenizer.js:60]: Oh, I see - no the source of text - was it typed manually or uploaded from the file does not refered to each other.
This conversion we need to convert source text from textarea to binary file - to send to remote alpheios tokenization service.

File: src/lib/tokenizers/alpheios-remote-tokenizer.js (hunk 2 above)
Comment (irina060981) [reply src/lib/tokenizers/alpheios-remote-tokenizer.js]: I would it this description to the issue - https://github.com/alpheios-project/alignment-editor-new/issues/32

**Pull request merged**